doc/flame/collision_detection.md

```diff
@@ -9,129 +10,169 @@ If you have the following needs you should at least consider to use
 
 Do note that the built-in collision detection system does not take collisions between two hitboxes
 that overshoot each other into account, this could happen when they either move too fast or `update`
 being called with a large delta time (for example if your app is not in the foreground). This
 behaviour is called tunneling, if you want to read more about it.
 
-Also note that the collision detection system doesn't work properly if you scale ancestors of the
-component that is `Collidable`.
+Also note that the collision detection system has a limitation that makes it not work properly if
+you have certain types of combinations of flips and scales of the ancestors of the hitboxes.
+
+
+## Adding hitboxes
+
+The `HitboxShape`s are normal components, so you add them to the component that you want to add
+hitboxes to just like any other component:
+
+`component.add(HitboxRectangle());`
+
+If you don't add any arguments to the hitbox, like above, the hitbox will try to fill its parent as
+much as possible. Except for having the hitboxes trying to fill their parents, there are two ways to
+initiate hitboxes and it is with the normal constructor where you define the hitbox by itself, with
+a size and a position etc. The other way is to use the `fromNormals` constructor which defines the
+hitbox in relation to the size of its parent. 
+
+You can read more about how the different shapes (and hitboxes) are defined in the
+[](#ShapeComponents) section.
+
+Remember that you can add as many `HitboxShape`s as you want to your `PositionComponent` to make up
+more complex areas. For example a snowman with a hat could be represented by three `HitboxCircle`s
+and two `HitboxRectangle`s as its hat.
+
 
 ## Mixins
-### HasHitboxes
-The `HasHitboxes` mixin is mainly used for two things; to make detection of collisions with other
-hitboxes and to more accurately recognize gestures on top of your `PositionComponent`s. Say that you
-have a fairly round rock as a `SpriteComponent` for example, then you don't want to register input
-that is in the corner of the image where the rock is not displayed, since an image is always
-rectangular. Then you can use the `HasHitboxes` mixin to define a more accurate polygon (or another
-shape) for which the input should be within for the event to be registered on your component.
-
-You can add new shapes to the `HasHitboxes` just like they are added in the below `Collidable`
-example.
-
-### Collidable
-The `Collidable` mixin is added to a `PositionComponent` that has a `HasHitboxes` and it is used for
-detecting collisions with other `Collidable`s. If you do not add a shape to your `HasHitboxes` 
-component it will never collide with anything. If you want the component to have a default 
-rectangular shape that fills the size of your component you can simply do 
-`addHitbox(HitboxRectangle())`.
-
-To make your component collidable you would start off something like this:
 
+### GestureHitboxes
```

Comment:
The whole document should maybe have another name, since the `GestureHitboxes` are dependant on hitboxes I think it fits in here, but maybe it should be under the gestures section and link to here instead.

Reply:
Moved it to `gesture-input.md`.